### PR TITLE
PytorchDataTeacher Multitask Functionality

### PR DESCRIPTION
**Pull request by klshuster**
Multitasking functionality for `PytorchDataTeacher`

To use, simply specify multiple tasks with a comma after `-pyt`, `-pytd`, or both.

E.g.: `py display_data.py -pyt squad,babi` or `py display_data.py -pytd vqa_v1,flickr30k`

### CODE REVIEW

**Comment by klshuster** [comment]
Note - this is failing a test on an unrelated diff (something to do with a squad teacher?)

**Comment by alexholdenmiller** [comment]
@emilydinan you have a flake error in the diff you landed

**Comment by emilydinan** [comment]
@klshuster this should be fixed if you rebase

**Comment by klshuster** [comment]
Thanks @emilydinan

File: parlai/core/pytorch_data_teacher.py [hunk 1]
```diff
@@ -306,6 +378,53 @@ def num_examples(self):
...
+        self.opt = opt
+        self.datatype = opt.get('datatype')
+        self.datafile = build_data(self.opt)
+        self._setup_data()
+        self.length_datafile = self.datafile + ".length"
+        self.training = self.datatype.startswith('train')
+        self._load_lens()
+
+    def __getitem__(self, index):
+        index %= self.num_episodes()
```
[the remark this answers is not in the document]
Comment (klshuster) [reply parlai/core/pytorch_data_teacher.py]: this used to not be the case given how I handled multiple epochs, but in the current/new implementation it is (I forgot to change it here)

File: parlai/tasks/coco_caption/agents.py [hunk 2]
```diff
@@ -243,7 +243,7 @@ def submit_load_request(self, image_id):
 
     def get(self, episode_idx, entry_idx=0):
         action = {
-            'text': "",
+            'text': "Please leave a Caption",
```
[the remark this answers is not in the document]
Comment (klshuster) [reply parlai/tasks/coco_caption/agents.py]: yeah this is changed now, probably why this has conflicts with master

File: parlai/core/pytorch_data_teacher.py [hunk 3]
```diff
@@ -306,6 +378,53 @@ def num_examples(self):
...
+
+
+class ParlAIConcatDataset(ConcatDataset):
+    """Override to set num_eps and num_exs"""
+
+    def num_episodes(self):
+        return sum(d.num_episodes() for d in self.datasets)
+
+    def num_examples(self):
+        return sum(d.num_examples() for d in self.datasets)
```
[the remark this answers is not in the document]
Comment (klshuster) [reply parlai/core/pytorch_data_teacher.py]: They should not change, I'll add that

File: parlai/tasks/coco_caption/agents.py [hunk 4]
```diff
@@ -322,6 +322,7 @@ def submit_load_request(self, image_id, split=None):
 
     def get(self, episode_idx, entry_idx=0):
         action = {
+            'text': '',
```
Comment (alexholdenmiller) [inline parlai/tasks/coco_caption/agents.py]: actually another nit: I'd prefer omitting text completely rather than putting an empty string

**Pull request merged**